Stripe/STPAPIClient+ApplePay.m

```diff
@@ -11,25 +11,26 @@
 #import "PKPayment+Stripe.h"
 #import "STPAPIClient+Private.h"
 #import "STPAnalyticsClient.h"
+#import "STPFraudSignals.h"
+#import "STPFormEncoder.h"
 
 FAUXPAS_IGNORED_IN_FILE(APIAvailability)
 
 @implementation STPAPIClient (ApplePay)
 
 - (void)createTokenWithPayment:(PKPayment *)payment completion:(STPTokenCompletionBlock)completion {
-    [self createTokenWithData:[self.class formEncodedDataForPayment:payment]
+    NSDictionary *metrics = [[STPFraudSignals sharedInstance] serialize];
+    NSData *data = [self.class formEncodedDataForPayment:payment metrics: metrics];
+    [self createTokenWithData:data
                    completion:completion];
 }
 
 #pragma clang diagnostic push
 #pragma clang diagnostic ignored "-Wdeprecated"
-+ (NSData *)formEncodedDataForPayment:(PKPayment *)payment {
++ (NSData *)formEncodedDataForPayment:(PKPayment *)payment metrics:(NSDictionary *)metrics {
```

Comment:
This makes me think that we've chosen the wrong abstraction in `STPAPIClient`. Ideally, all of these extensions would need no knowledge of `STPFraudMetrics`. They need this because `STPAPIClient` currently uses `NSData` to represent data that is ready to go over the wire. I think this is wrong. I think it should instead use an `NSDictionary` of key pairs that it then internally converts to an `NSData` and sends to the API. This would allow it to add things like additional metrics seamlessly under the hood (it could also do this now by deserializing the NSData, modifying it, and reserializing it, but that would be horrible.)

This would obviously need to be a separate change, and would need to be extremely well-tested, but I think would be the better way to approach changes like this in the future. I'd be bullish on making that change and then rebasing this PR off of it.